src/main/scala/com/kifi/franz/SimpleSQSQueue.scala

```diff
@@ -15,5 +16,10 @@ class SimpleSQSQueue(protected val sqs: AmazonSQSAsync, val queue: QueueName, pr
 
 class JsonSQSQueue(protected val sqs: AmazonSQSAsync, val queue: QueueName, protected val createIfNotExists: Boolean = false) extends SQSQueue[JsValue] {
   protected implicit def asString(json: JsValue) = Json.stringify(json)
-  protected implicit def fromString(s: String) = Json.parse(s)
+  protected implicit def fromString(s: String) = {
```

Comment:
I don't think it is semantically a good idea to repurpose `JsUndefined` as an error container since that might already be used in some other way in the client code using the queue (even if, of course, it can't be in the messages) causing unintended downstream errors.
I think a better - if more involved - way of achieving the same end would be to create a `SafeSQSQueue[T]` that is basically a `SQSQueue[Try[T]]` (not quite since inputs are still `T` not `Try[T]`) and then derive a `SafeJsonSQSQueue` from that.
That would actually be a really nice addition to this library, imho.